src/raw.rs

```diff
@@ -363,3 +363,11 @@ pub fn string_append_buffer(
             .into()
     }
 }
+
+pub fn subscribe_to_server_event(
+    ctx: *mut RedisModuleCtx,
+    event: RedisModuleEvent,
+    callback: RedisModuleEventCallback,
+) -> Status {
+    unsafe { RedisModule_SubscribeToServerEvent.unwrap()(ctx, event, callback).into() }
+}
```

Comment:
I suggest adding a higher level wrapper above this layer.
Have you seen the code in `src/context/timer.rs`? It can be done in the same way.

Reply:
The thing here is that `RedisModuleEventCallback` receives a pointer to a struct with extra data about the event where the struct type changes based on which event we chose . So Wapping this nicely means we need to be aware of the types of events that exist and then mapping the `data` type to the correct struct type when when calling the high level callback.
I'm willing to attempt to do this but it means we're more tightly bount between our rust code and redisumode.h, when new events are added it won't work out of the box, we'll need to update our translation layer.
Let me know if I was clear here.

Reply:
Another issue is that there's no user data here, so there's no place to store the high level callback.

Reply:
Ok, these are all good points. I mainly wanted to bring these up, not sure they're necessary.
Let's do this on an as-needed basis and see where it leads; for now I see to good reason to invest in this.